### ISSUE TITLE
what should be the value of PYNGUIN_DANGER_AWARE

### ISSUE THREAD

**Issue by hackertron** [se2p/pynguin#16]
The docs says PYNGUIN_DANGER_AWARE should be set but what value ? It's not specified anywhere. I have tried True, "True", true

**Comment by Wooza** [comment]
Hi,

the value does not matter, it is enough that the environment variable exists.
But I want to remind you that using Pynguin outside of a container, e.g. Docker, can be dangerous, depending on what kind of code you use it on. So please take a look at the [quick start guide](https://pynguin.readthedocs.io/en/latest/user/quickstart.html).

Best regards
Florian

**Comment by stephanlukasczyk** [comment]
We'll update the documentation regarding this excellent question. Thank you @hackertron for the question and @Wooza for the quick answer.

**Comment by Geveltekens** [comment]
Can you be a bit more specific on what the user/admin should do? Something like the following before running Pynguin?:
```
$ export PYNGUIN_DANGER_AWARE
$ PYNGUIN_DANGER_AWARE="yes"
$ printenv PYNGUIN_DANGER_AWARE
yes
```
Is something else required?

**Comment by stephanlukasczyk** [comment]
As I've also elaborated in an StackOverflow answer related to this variable (see https://stackoverflow.com/a/70302516/4293396), the value does not matter.  Pynguin just checks that the variable is defined, nothing else.  Thus,
`export PYNGUIN_DANGER_AWARE=` should be sufficient.  However, you can use arbitrary values, personally, I used to have `export PYNGUIN_DANGER_AWARE=1`.

**Comment by Geveltekens** [comment]
Ok, thank you. I had indeed seen that SO response.

Can you update the documentation to state how environment variables can be set? I had to look up what an environment variable was and made an educated guess to set it.

I am still confused by this functionality. If one sets PYNGUIN_DANGER_AWARE="no" or PYNGUIN_DANGER_AWARE="False", then we have a situation akin to the [Liar's paradox](https://en.wikipedia.org/wiki/Liar_paradox). Hence, why have the feature at all?

**Comment by stephanlukasczyk** [comment]
>Can you update the documentation to state how environment variables can be set? I had to look up what an environment variable was and made an educated guess to set it.

We could but the problem is that it is done differently in different operating systems. On Linux/macOS/Unix one can always set such a variable temporarily on the command line using `export VAR_NAME=value`. To set it permantently, you might want to add it to some `~/.profile` or whatsoever file. Windows is a whole different story, were one is force to dig deep in the depth of the system configuration dialogues (at least that's what it was a decade ago when I've used Windows the last time).

TL;DR: Putting this in the documentation will never cover all cases, thus I believe it's easier when users just Google for how it's done on their OS.

>I am still confused by this functionality. If one sets PYNGUIN_DANGER_AWARE="no" or PYNGUIN_DANGER_AWARE="False", then we have a situation akin to the [Liar's paradox](https://en.wikipedia.org/wiki/Liar_paradox). Hence, why have the feature at all?

The reason is explained in the documentation: Pynguin will execute the module under test, which can cause harm to your system.

Consider the following example
```python
import shutils

def foo():
   shutils.rmtree("/")
```
Calling `foo()` will attempt to recursively delete all files and folders from your file system (assuming you are running Linux).  If you would execute Pynguin as root user (which you should never do) on such a module, you hopefully have a good backup...

The intention of this was to prevent stupid mistakes that happen in hectic. The value does not matter, Pynguin will never check it. But by defining the environment variable, you tell Pynguin that you know what you are doing (or at least that there might be a risk that the execution of Pynguin will whipe your hard disk). And it's a safeguard for us: if you set the variable and executed Pynguin you cannot blame us for any harm that was caused by this action.

**Comment by Geveltekens** [comment]
I already understood the intention and the sentiment. Stating "prevent stupid mistakes" and "it's a safeguard for us" are meaningless arguments. It is a superficial barrier to entry. You are asking your users to jump through a hoop, but are only explaining what the hoop is called, barely how to jump through it.

**Comment by stephanlukasczyk** [comment]
What are you suggesting to do then? Remove the check for the environment variable? That's as easy as removing the following condition https://github.com/se2p/pynguin/blob/083c49677c55b5f94af619c25f727b5ff6ed8d63/src/pynguin/cli.py#L167-L175

But will it be a good change?  Some might agree, some might disagree.  Thus, as the tool maintainer I decide to keep the check and force “users to jump through a hoop”, as you've framed it.
